src/palimpzest/query/optimizer/rules.py

```diff
@@ -325,6 +349,8 @@ def substitute(cls, logical_expression: LogicalExpression, **physical_op_params)
             )
             physical_expressions.append(expression)
 
```

Comment:
should we also do the `deduped_physical_expressions` pattern that is used below here?